### PR TITLE
Fileio/open_c close_c

### PR DESCRIPTION
**Pull request by WaelYoussfi**

### CODE REVIEW

**Comment by WaelYoussfi** [comment]
@maxteufel We can now open and close streams in C and get the return result as a tuple.
`open_source` returns `tuple fd errno` in success and `tuple nil errno` in failure.
`close_source` returns `tuple bool errno`.

File: src/dev/flang/be/c/Intrinsics.java [hunk 1]
```diff
@@ -202,6 +202,39 @@ public class Intrinsics extends ANY
...
+              ),
+            openresults.index(CExpr.ident("1")).assign(new CIdent("errno").castTo("fzT_1i64"))
+            );
+        }
+        );
+    put("fuzion.std.fileio.on_close"   , (c,cl,outer,in) ->
+        {
+          return CStmnt.seq(
+            CStmnt.iff(CExpr.call("fclose", new List<>(A0.castTo("FILE *"))).eq(CExpr.int8const(0)), c._names.FZ_TRUE.ret()),
+            c._names.FZ_FALSE.ret()
```
[the remark this answers is not in the document]
Comment (WaelYoussfi) [reply src/dev/flang/be/c/Intrinsics.java:225]: I want to avoid mutations whenever possible.

[the remark this answers is not in the document]
Comment (michaellilltokiwa) [reply src/dev/flang/be/c/Intrinsics.java:225]: I think if `fclose` returns `EOF` `errno` is always set. So we can just discard the result of `fclose'` in that case?

File: lib/fuzion/std/fileio.fz [hunk 2]
```diff
@@ -201,11 +201,11 @@ public fileio is
   # in case of failure returns tuple nil i64 where the second parameter represents the error number 
   # NYI : return type might be modified to a choice of generics instead
   #
-  public open_source(
+  public open_c(
```
[the remark this answers is not in the document]
Comment (WaelYoussfi) [reply lib/fuzion/std/fileio.fz:204]: That looks convenient.

File: lib/fuzion/std/fileio.fz [hunk 3]
```diff
@@ -195,3 +195,46 @@ public fileio is
                  path Object,
                  # dummy parameter to enable overload
                  _ unit) bool is intrinsic
+
+  # Opens an IO source using a path and returns
```
Comment (michaellilltokiwa) [inline lib/fuzion/std/fileio.fz:199]: returns and returns

**Comment by WaelYoussfi** [comment]
see #694

**Pull request closed without merging**